docs/architecture/commitlog/fileformat.rst

```diff
@@ -0,0 +1,63 @@
+ScyllaDB commitlog segment file format
+======================================
+
+**Note:** Commitlog file formats are subject to change between scylla versions. Users should not make assumptions about nor rely on them.
+Commitlog files should *never* be used across ScyllaDB updates. This information is provided mainly for ScyllaDB contributors.
+
+File descriptor structure
+.........................
+
+ScyllaDB commitlog segment files are named with a versioned, time-indexed scheme, as::
```

Comment:
Maybe you can start with an even more general overview of what is a commit log (though maybe this is overkill, and unnecessary), and what is a commit log "segment" and why those exist and what determines their size, etc.? Or maybe this is documented elsewhere?

Reply:
Sure. At some point. I only added this because @avikivity requested a doc on the segment file format.

Reply:
Moved to dev folder. Doh. Managed to miss that it still exists. Makes much more sense there. Thanks.